File: leetcode/0013-roman-to-integer/romanToInt.go
package leetcode

func romanToInt(s string) int {
	var romanIntMap = map[string]int{
		"I":  1,
		"V":  5,
		"X":  10,
		"L":  50,
		"C":  100,
		"D":  500,
		"M":  1000,
		"IV": 4,
		"IX": 9,
		"XL": 40,
		"XC": 90,
		"CD": 400,
		"CM": 900,
	}

	var int = 0
	for index := 0; index < len(s); index++ {
		var roman = string(s[index])
		var combinRoman string
		if index+1 == len(s) {
			combinRoman = roman
		} else {
			combinRoman = roman + string(s[index+1])
		}

		if value, ok := romanIntMap[combinRoman]; ok {
			int = int + value
			index++
			continue
		}

		int = int + romanIntMap[roman]

	}

	return int
}

File: leetcode/0013-roman-to-integer/romanToInt.ts
function romanToInt(s: string): number {
  var romanIntMap: Record<string, number> = {
    I: 1,
    V: 5,
    X: 10,
    L: 50,
    C: 100,
    D: 500,
    M: 1000,

    IV: 4,
    IX: 9,
    XL: 40,
    XC: 90,
    CD: 400,
    CM: 900,
  };

  let int = 0;
  for (let index = 0; index < s.length; index++) {
    const roman = s[index]!;
    const nextRoman = s[index + 1] ?? "";
    const combineRoman = `${roman}${nextRoman}`;

    if (romanIntMap[combineRoman] != undefined) {
      int = int + romanIntMap[combineRoman]!;

      index++;

      continue;
    }

    int = int + romanIntMap[roman]!;
  }

  return int;
}

export default romanToInt;
